cnrancher/octopus
# feat: add mqtt adaptor

add MQTT adaptor logic
address #47




**Fixes:**
#47 


**Problem:**
In our planning, there may be some vendors that have devices that support the MQTT protocol, and we need to support this so that these types of devices can quickly access the octopus.


**Solution:**
So our solution for devices that support the native MQTT protocol is to use the MQTT adaptor, develop the MQTT adaptor to subscribe to the state that occurs for devices that support the native MQTT protocol, and distribute the user's instructions from octopus through the MQTT adaptor.


**Test plan:**
1. Create an MQTT publisher to simulate a device that supports the MQTT protocol, and use it to send messages to the corresponding topics
2. use all_in_on.yaml and dl_mqtt.yaml in the e2e directory to simulate the creation of a corresponding device from octopus, and configure the spec field of dl_mqtt.yaml
3. observe whether the status of the MqttDevice resource in k3s has changed, as a way to simulate the state of the device synchronization to octopus.
